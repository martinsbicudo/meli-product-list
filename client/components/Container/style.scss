.Container {
  display: flex;
  margin: auto;
  width: 100%;
  max-width: map-get($grid, "container");
  padding: 0 2rem;
  box-sizing: border-box;
  min-height: fit-content;
}
