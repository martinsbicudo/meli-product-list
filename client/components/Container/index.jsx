import React from 'react';
import { string, node, oneOfType } from 'prop-types';
import classnames from 'classnames';
import style from './style.scss';

const Container = ({
  className,
  children,
}) => (
  <div
    className={classnames(
      style.Container,
      className,
    )}
  >
    {children}
  </div>
);

Container.propTypes = {
  className: string,
  children: oneOfType([string, node]).isRequired,
};

Container.defaultProps = {
  className: '',
};

export default Container;
