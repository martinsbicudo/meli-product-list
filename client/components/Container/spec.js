import Container from '.';

describe('Components - <Container />', () => {
  const testMessage = 'TEST_CHILDREN';
  const createWrapper = () => render(
    <Container>
      <span>{testMessage}</span>
    </Container>,
  );

  test('Should finds Children', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessage)).toBeInTheDocument();
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
