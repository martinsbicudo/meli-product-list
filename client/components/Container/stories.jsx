import React from 'react';
import { storiesOf } from '@storybook/react';
import Container from '.';

storiesOf('Container', module)
  .add('With Children', () => (
    <Container>
      <span>Children Container</span>
    </Container>
  ));
