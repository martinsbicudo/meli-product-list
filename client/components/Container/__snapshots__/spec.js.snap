// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - <Container /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container"
  >
    <span>
      TEST_CHILDREN
    </span>
  </div>
</DocumentFragment>
`;
